src/lib/storage/index/group_key/group_key_index.cpp

```diff
@@ -14,41 +15,60 @@ size_t GroupKeyIndex::estimate_memory_consumption(ChunkOffset row_count, ChunkOf
 }
 
 GroupKeyIndex::GroupKeyIndex(const std::vector<std::shared_ptr<const BaseSegment>>& segments_to_index)
-    : BaseIndex{get_index_type_of<GroupKeyIndex>()},
-      _indexed_segments(std::dynamic_pointer_cast<const BaseDictionarySegment>(segments_to_index[0])) {
-  Assert(static_cast<bool>(_indexed_segments), "GroupKeyIndex only works with dictionary segments_to_index.");
+    : AbstractIndex{get_index_type_of<GroupKeyIndex>()},
+      _indexed_segment(segments_to_index.empty()
+                           ? nullptr
+                           : std::dynamic_pointer_cast<const BaseDictionarySegment>(segments_to_index[0])) {
+  Assert(static_cast<bool>(_indexed_segment), "GroupKeyIndex only works with dictionary segments_to_index.");
   Assert((segments_to_index.size() == 1), "GroupKeyIndex only works with a single segment.");
 
-  // 1) Initialize the index structures
-  // 1a) Set the index_offset to size of the dictionary + 1 (plus one to mark the ending position)
-  //     and set all offsets to 0
-  _index_offsets = std::vector<size_t>(_indexed_segments->unique_values_count() + 1u, 0u);
-  // 1b) Set the _index_postings to the size of the attribute vector
-  _index_postings = std::vector<ChunkOffset>(_indexed_segments->size());
+  // 1) Initialize the index_offset:
+  //    Set the index_offset to size of the dictionary + 2 (plus one for null and plus one to mark the ending position)
+  //    and set all offsets to 0.
+  //    Using `_index_offsets`, we want to count the occurences of each ValueID of the attribute vector (AV).
+  //    The ValueID for null in an AV is the highest available ValueID in the dictionary + 1.
+  //    `unique_values_count` returns the size of dictionary which does not store a ValueID for null.
+  //    Therefore we have `unique_values_count` + 1 (for null) ValueIDs for which we want to count the occurrences.
+  _index_offsets = std::vector<ChunkOffset>(
+      _indexed_segment->unique_values_count() + 1u /*for null*/ + 1u /*to mark the ending position */, 0u);
```

Comment:
Add a comment explaining that albeit the NULL values are stored in a different data structure in the super class, we first collect them together with the other values and cut them off later?